openapi.yml

```
@@ -31,81 +54,162 @@ paths:
       type: string
       example: "6810e776880c02933d47db1b9fc05908e5386b96"
     TokenAmount:
-      description: Amount of a token. uint112 encoded in decimal.
+      description: Amount of a token. uint256 encoded in decimal.
       type: string
       example: "1234567890"
-    Nonce:
-      description: Nonce of the batch in which the order is valid. uint32.
-      type: integer
-    OrderNew:
+    FeeInformation:
+      description: |
+        Provides the information to calculate the fees. 
+      type: object
+      properties:
+        expirationDate:
+          description: |
+            Expiration date of the offered fee. Order service might not accept 
+            the fee after this expiration date. Encoded as ISO 8601 UTC.
+          type: string
+          example: "2020-12-03T18:35:18.814523Z"
+        minimalFee:
+          description: Absolute amount of minimal fee charged per order in specified sellToken
+          $ref: "#/components/schemas/TokenAmount"
+        feeRatio:
+          description: The fee ratio charged on a sellAmount. Denoted in basis points
+          example: 10
+          type: number
+          format: int32
+    OrderType:
+      description: Is this a buy order or sell order?
+      type: string
+      enum: [buy, sell]
+    OrderCreation:
       description: Data a user provides when creating a new order.
       type: object
       properties:
+        sellToken:
+          description: "ERC20 token to be sold"
+          $ref: "#/components/schemas/Address"
         buyToken:
+          description: "ERC20 token to be bought"
           $ref: "#/components/schemas/Address"
-        buyAmount:
-          $ref: "#/components/schemas/TokenAmount"
         sellAmount:
+          description: "Amount of sellToken to be sold in atoms"
           $ref: "#/components/schemas/TokenAmount"
-        sellToken:
-          $ref: "#/components/schemas/Address"
-        sellTokenTip:
+        buyAmount:
+          description: "Amount of buyToken to be bought in atoms"
           $ref: "#/components/schemas/TokenAmount"
-        orderType:
-          description: Is this a buy order or sell order?
-          type: string
-          enum: [buy, sell]
-        fillType:
-          description: Is this a fill-or-kill order a partially fillable order?
-          type: string
-          enum: [fillorkill, partial]
-        nonce:
-          $ref: "#/components/schemas/Nonce"
         validTo:
-          description: Time offset in seconds from epoch until which the order is valid. uint32.
+          description: Unix timestamp until the order is valid. uint32.
+          type: integer
+        appData:
+          description: |
+            Arbitrary identifier sent along with the order.
+            Could be used to track the interface or other meta-aspects of the order. uint32 encoded
           type: integer
+        feeAmount:
+          description: "Fees: feeRatio + minimal_fee in atoms"
```

@josojo I understand it should be

```suggestion
          description: "Fees: feeRatio * sellAmount + minimal_fee in atoms"
```